mmv1/third_party/terraform/services/compute/resource_compute_instance_group_manager.go.erb

```diff
@@ -1322,22 +1322,58 @@ func flattenStatefulPolicyStatefulInternalIps(statefulPolicy *compute.StatefulPo
 	return result
 }
 
-func flattenStatefulPolicyStatefulExternalIps(statefulPolicy *compute.StatefulPolicy) []map[string]interface{} {
+func flattenStatefulPolicyStatefulExternalIps(d *schema.ResourceData, statefulPolicy *compute.StatefulPolicy) []map[string]interface{} {
 	if statefulPolicy == nil || statefulPolicy.PreservedState == nil || statefulPolicy.PreservedState.ExternalIPs == nil {
 		return make([]map[string]interface{}, 0, 0)
 	}
-	result := make([]map[string]interface{}, 0, len(statefulPolicy.PreservedState.ExternalIPs))
+
+	initialResult := make([]map[string]interface{}, 0, len(statefulPolicy.PreservedState.ExternalIPs))
 	for interfaceName, externalIp := range statefulPolicy.PreservedState.ExternalIPs {
 		data := map[string]interface{}{
 			"interface_name": interfaceName,
 			"delete_rule": externalIp.AutoDelete,
 		}
 
-		result = append(result, data)
+		initialResult = append(initialResult, data)
 	}
-	return result
+
+	// statefulPolicy.PreservedState.ExternalIPs is affected by API-side reordering of external IPs, where ordering
+	// is done by the interface_name value. Below we intend to reorder the external IPs to match the order in the config.
+	// Any external IPs found from the API response that aren't in the config are appended to the end of the list.
+
+	val, ok := d.GetOk("stateful_external_ip")
+	configIpOrder := val.([]interface{})
+	if !ok || len(configIpOrder) == 0 {
+		return initialResult // return information from the API unchanged
+	}
+
+	order := map[string]int{} // record map of interface name to index
+	for i, el := range configIpOrder {
+		externalIp := el.(map[string]interface{})
+		name := externalIp["interface_name"].(string)
+		order[name] = i
+	}
+
+	orderedResult := make([]map[string]interface{}, len(configIpOrder))
+	unexpectedIps := []map[string]interface{}{}
+	for _, el := range initialResult {
```

Comment:
Looking at this with fresh eyes, I don't think I need to have `initialResult` (that contains the data transformed map => slice) and also have the separate process to create `orderedResult` (that's produced by re-ordering `initialResult`). Instead I could re-order the data and convert it from map => slice at the same time.

Which would be easier to understand?

Reply:
Refactored in https://github.com/GoogleCloudPlatform/magic-modules/pull/9577/commits/65785828b58fc8f57e788995a0ac211efa5418ab but I'm open to reverting that commit